### ISSUE TITLE
global time series plot has issues with ocean plots when simulation is beyond the specified ending year

### ISSUE THREAD

**Issue by tangq** [E3SM-Project/zppy#256]
The cdscan command called when creating the global time series plot will include years beyond the specified ending year for ocean. This will cause different lengths of years and ocean variables (e.g., ohc and volumn) and hence errors when plotting. We used to have the similar issue for atmospheric variables and it has been solved (But I couldn't find that PR). This ocean issue can be fixed in a similar way.

An example with this issue can be found at: /global/cfs/projectdirs/e3sm/tang30/E3SMv2/v2.NARRM.historical_0201/post/scripts

I have to hack the ocean variable index to bypass it temporarily.

**Comment by tangq** [comment]
@forsyth2, can you try again?

**Comment by forsyth2** [comment]
@tangq Thanks, I can access now. What's the path to your `cfg` file though? Or can you paste it here?

**Comment by tangq** [comment]
The cfg files are at /global/homes/t/tang30/E3SM_scripts/post. Thanks.

**Comment by golaz** [comment]
I ran into a similar issue. I believe this is fixed in https://github.com/E3SM-Project/zppy/pull/268. We still had some hard-coded segment lengths of 10 years.

**Comment by forsyth2** [comment]
@golaz Thanks, does this mean this is the same issue as #208 (and the PR #219)?

**Comment by tangq** [comment]
That's right, @forsyth2 .

**Comment by forsyth2** [comment]
@tangq Have you run this using `zppy` since #268 was merged? Does it work now?

**Comment by tangq** [comment]
I haven't tried it yet. Also, #268 is not in the current e3sm unified env, I believe.

**Comment by forsyth2** [comment]
That's correct, it will be included in tomorrow's `zppy` release candidate `rc3` and the subsequent Unified release candidate. In the meantime, you can create a `dev` environment: https://e3sm-project.github.io/zppy/_build/html/main/getting_started.html#b-development-environment
